### PR TITLE
Hide Softlayer credentials in error message

### PR DESCRIPTION
**Pull request by zhanggbj**
Hi Dr.Max @maximilien,

This commit it to hide Softlayer credentials in error message. Please help to take a review. Thanks.
[#140904659](https://www.pivotaltracker.com/story/show/140904659)

### CODE REVIEW

**Comment by maximilien** [comment]
@zhanggbj Travis is not passing. Please fix that.

**Comment by zhanggbj** [comment]
@maximilien , it's a build problem. Re-triggered and it's `passed` now. Thanks.

File: client/softlayer_client_test.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ var _ = Describe("SoftLayerClient", func() {
 			if urlErr, ok := err.(*url.Error); ok {
 				err = urlErr.Err
 			}
-			Expect(err).To(Equal(errDialFailed))
+			Expect(err.Error()).To(ContainSubstring("dial failed"))
```
Comment (maximilien) [inline client/softlayer_client_test.go]: Why are we only looking for `"dail failed"` as a test for this change? I would expect a new test that shows that a error with message that includes API key is masked, so a text like: `"12345abcd@softlayer.com"` changes to `"************@api.softlayer.com"`.

Please add such a test and I will merge. Thanks.

**Comment by maximilien** [comment]
@zhanggbj see one last comment. Thx.

File: client/softlayer_client_test.go (hunk 1 above)
Comment (zhanggbj) [inline client/softlayer_client_test.go]: Added one more expect in this test case. Thanks.

**Comment by zhanggbj** [comment]
@maximilien , comments addressed, please take another review. Thank you.

**Pull request merged**